### PR TITLE
Fix overlap integrals for complex beams

### PR DESCRIPTION
**Pull request by chrisbrahms**
This delightful little feature cost me several hours today: when `dot`ing two vectors containing complex values, the first is **automatically** conjugated:
```julia
help?> dot
search: dot @__dot__ stdout DEPOT_PATH adjoint fieldcount codepoint fieldoffset isdisjoint denominator modifyproperty! StridedVecOrMat DenseVecOrMat redirect_stdout DimensionMismatch

  dot(x, y)
  x ⋅ y

  Compute the dot product between two vectors. For complex vectors, the first vector is conjugated.
```

This PR removes additional manual conjugates to fix overlap integrals for beams which evaluate to complex-valued vectors (e.g. diverging Gaussian beams using `Fields.gauss_beam`). It also splits the integrals into real and imaginary parts to allow for potentially complex overlap integral values (ie a phase between the mode and the beam).

### CODE REVIEW

File: src/Modes.jl [hunk 1]
```diff
@@ -322,11 +322,13 @@ coordinates `xs = (r,θ)`.
 """
 function overlap(m::AbstractMode, E)
     dl = dimlimits(m)
-    function f(xs)
-        0.5*sqrt(ε_0/μ_0)*dot(conj(Exy(m, xs)), E(xs))*xs[1]
+    rval, _ = hcubature(dl[2], dl[3]; maxevals=1000) do xs
```
Comment (jtravs) [inline src/Modes.jl:325]: Does `hcubature` not handle the complex case? The C library does.

File: test/test_modes.jl [hunk 2]
```diff
@@ -143,6 +143,27 @@ Emn2 = Eωm2/norm(Eωm2)
 @test all(isapprox.(En2, Emn2, atol=1e-3*maximum(abs.(Emn2))))
 end
 
+##
+@testset "overlap with Gaussian beam (function)" begin
```
Comment (jtravs) [inline test/test_modes.jl:147]: There is a testset in `test_fields.jl` that covers the non-diverged case.

**Review by jtravs — APPROVED** [review]
LGTM, but see minor comments.

File: test/test_modes.jl (hunk 2 above)
Comment (chrisbrahms) [reply test/test_modes.jl:147]: I'd forgotten about that. Will move this one there

File: src/Modes.jl (hunk 1 above)
Comment (chrisbrahms) [reply src/Modes.jl:325]: It doesn't see https://github.com/JuliaMath/Cubature.jl/issues/21 As far as I can tell the C library only supports it through using the vector interface, but maybe I'm wrong? Anyway here we can use the pure Julia implementation instead and it works fine without splitting.

Comment (jtravs) [reply src/Modes.jl]: Yes, the vector interface but with pairwise error. Anyway, the Julia solution is better.

**Pull request merged**